faculty_mill/publish.py

```diff
@@ -0,0 +1,94 @@
+import os
+from pathlib import Path
+from typing import Optional
+from uuid import UUID
+from time import sleep
+import click
+import sherlockml
+import sml.auth
+
+
+def publish(
+    report_name: str,
+    path: Path,
+    show_code: bool = False,
+    report_id: Optional[UUID] = None,
+    project_id: Optional[UUID] = None,
+    user_id: Optional[UUID] = None,
+):
+    """Publish a notebook as a report.
+
+    Parameters
+    ----------
+    report_name : str
```

Comment:
Is there a reason to include the types in the docstring if we are typing the variables?

Reply:
I think for discoverability?

Reply:
What do you mean?

Reply:
Some people might only look at the doc string to find the types, so this would help them

Reply:
Fair enough